recipes/tests.py

```diff
@@ -288,3 +286,63 @@ def test_admin_can_destroy_recipe(self):
             reverse('recipe-detail', kwargs={'pk': recipe.id})
         )
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+
+class LikeDetailTestCase(BaseTestCase):
+    def create_like(self, user, recipe):
+        return Like.objects.create(user=user, recipe=recipe)
+
+    def test_user_can_like_recipe(self):
+        user, client = self.create_user()
+        recipe = self.create_recipe(user=user)
+        response = client.post(
+            reverse('like-recipe', kwargs={'pk': recipe.id})
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIsNotNone(recipe.likes.first())
+
+    def test_non_user_can_like_recipe(self):
+        user, user_client = self.create_user()
+        client = APIClient()
+        recipe = self.create_recipe(user=user)
+        response = client.post(
+            reverse('like-recipe', kwargs={'pk': recipe.id})
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_user_can_list_like_of_recipes(self):
+        user, client = self.create_user()
+        recipe = self.create_recipe(user=user)
+        response = client.get(
+            reverse('like-recipe', kwargs={'pk': recipe.id})
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_non_user_can_list_like_of_recipes(self):
+        user, user_client = self.create_user()
+        client = APIClient()
+        recipe = self.create_recipe(user=user)
+        response = client.get(
+            reverse('like-recipe', kwargs={'pk': recipe.id})
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_user_can_unlike_recipe(self):
+        user, client = self.create_user()
+        recipe = self.create_recipe(user=user)
+        self.create_like(user=user, recipe=recipe)
+        response = client.delete(
+            reverse('like-recipe', kwargs={'pk': recipe.id})
+        )
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertIsNone(recipe.likes.first())
```

Comment:
You need to send this request second time to check it runs as expected if there is no like.